lib/deck.rb

```diff
@@ -0,0 +1,47 @@
+class Deck
+  # will need this later, could probably find a much better way to do this
+  attr_accessor :cards
+
+  def initialize(cards = [])
+    @cards = cards
+  end
+
+  def rank_of_card_at(index)
+    @cards[index].rank
+  end
+
+  def high_ranking_cards
+    @cards.select { |card| card.rank >= 11 }
+  end
+
+  def percentage_high_ranking
+    self.high_ranking_cards.length.to_f / @cards.length.to_f
```

Comment:
Good job re-using the return value of the `high_ranking_cards` method here to keep your code DRY.